test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -97,9 +98,18 @@ def setup_sc(self, sched_id, partition, port):
         a = {'partition': partition,
              'sched_host': self.server.hostname,
              'sched_port': port}
+        if sched_priv is not None:
+            a['sched_priv'] = sched_priv
+        if sched_log is not None:
+            a['sched_log'] = sched_log
         self.server.manager(MGR_CMD_CREATE, SCHED, a, id=sched_id)
-        self.scheds[sched_id].create_scheduler()
-        self.scheds[sched_id].start()
+        if(len(a) > 3):
```

Comment:
If 'sched_priv' is None, but 'sched_log' is not None, len(a) > 3 will be satisfied and it will crash the test when os.path.dirname(sched_priv) gets executed. Maybe change this condition to "if 'sched_priv' in a" instead ?

Reply:
Thanks @agrawalravi90 for the valuable suggestion and have implemented the same.